modules/web_scraper.py

```diff
@@ -0,0 +1,238 @@
+    # Creating the date property
+    tweet_date = _get_tweet_date(tweet_element)
+    tweet_object["date"] = tweet_date
+    return tweet_object
+
+
+def _create_tweet_objects(soup: bs4.BeautifulSoup, hashtags):
+    """
+    Extracts tweets from given soup and returns an array of strings, each string containing a tweet.
+    """
+    tweet_elements = soup.find_all("table", class_=["tweet"])
+    tweets = []
+
+    for element in tweet_elements:
+        tweet_object = _create_tweet_object(element, hashtags)
+        tweets.append(tweet_object)
+    # Return result
+    return tweets
+
+
+def _extract_inner_text(tweet_element):
+    """
+    Extracts raw text from a tweet element.
+    """
+    # The content of the tweet is put into several divs so we find all those divs and extract their inner text
+    tweet_fractions = tweet_element.find_all("div", class_=["dir-ltr"])
+    tweet_text = ""
+    for element in tweet_fractions:
+        tweet_text += str(element.text)
+    # Create initial tweet object and return it
+    return tweet_text
+
+
+def _extract_data_urls(tweet_element):
+    """
+    Extracts the data-url attributes from a tweet element.
+    """
+    # Getting all <a>-tags
+    tweet_a_tags = tweet_element.find_all("a")
+    data_urls = []
+    for element in tweet_a_tags:
+        try:
+            # If the tag has an attribute "data-url" then we take the link from this URL
+            data_url = element["data-url"]
+            data_urls.append(data_url)
+        except:
+            pass
+    return data_urls
+
+
+def _get_tweet_date(tweet_element):
+    """
+    Returns a datetime object with the date contained in the tweet.\nThe minutes and hours etc. are not accurate.\n
+    If the tweet is from before 2020 it will still be returned as if it was from 2020 :Hide_the_pain_Harold: Who needs tweets from the previous year anyway
+    """
+    # Finding element that contains the timestamp
+    timestamp_element = tweet_element.find("td", class_=["timestamp"])
+    # Inner element with timestamp or date
+    timestamp_inner_element = timestamp_element.find("a")
+    # The date is stored as inner text
+    timestamp = timestamp_inner_element.text
+    # If the timestamp contains "min" or "h"(hour) etc. it means the tweet is from today so we just return the date today
+    if "min" in timestamp or timestamp[-1] == "m" or timestamp[-1] == "s" or "h" in timestamp or "now" in timestamp:
+        return "{},{},{}".format(datetime.now().date().year, datetime.now().date().month, datetime.now().date().day)
+    # Extract date to ensure proper formatting
+    tweet_date = datetime.strptime(timestamp + " 2020", "%b %d %Y")
+    # The weird looking return statement is made to make it easier for other parts of the program to read the date
+    return "{},{},{}".format(tweet_date.date().year, tweet_date.date().month, tweet_date.date().day)
+
+
+def _emoji_description_extractor(text: str):
+    """ 58 lol
```

Comment:
58 - The previous amount of lines in the function before optimization. Maybe I should clean up the comments and commit the changes..